src/x509.c

```diff
@@ -8634,9 +8634,15 @@ WOLF_STACK_OF(WOLFSSL_X509)* wolfSSL_X509_chain_up_ref(
                 XFREE(cert, NULL, DYNAMIC_TYPE_CERT);
                 return WOLFSSL_FAILURE;
             }
+   
+            cert->decodedCert = (DecodedCert*)XMALLOC(sizeof(DecodedCert), cert->heap,
```

Comment:
Please add NULL checking here in case the malloc fails.

Reply:
Null check added.